src/main/java/stringcalculator/StringSplitter.java

```diff
@@ -0,0 +1,49 @@
+package stringcalculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.stream.Collectors.toList;
+
+public class StringSplitter {
+    private static final List<String> DEFAULT_DELIMITER = Arrays.asList(",", ":");
+    private static final String DELIMTER = "|";
+    private static final String MATCH_SPECIAL = "\\?|\\.|\\$|\\|";
+    private static final String ADD_SPECIAL_DELIMITER = "\\";
+    private static final Pattern CUSTOM_PATTERN = Pattern.compile("//(.)\n(.*)");
+
+    private List<String> delimiters;
+
+    public StringSplitter() {
+        this(new ArrayList<>(DEFAULT_DELIMITER));
+    }
+
+    public StringSplitter(List<String> delimiters) {
+        this.delimiters = delimiters;
+    }
+
+    public List<String> split(String text) {
+        String checkedText = matchCustom(text);
+        String delimiter = String.join(DELIMTER, delimiters);
+        return Arrays.stream(checkedText.split(delimiter)).collect(toList());
```

Comment:
```suggestion
        return Arrays.stream(checkedText.split(delimiter)).collect(toList());
```
스트림을 사용하지 않고 구현할수 있지 않을까요 ?

Reply:
```suggestion
        return Arrays.asList(checkedText.split(delimiter));
```
로 가능할것 같습니다! 리팩토링 과정에서 map으로 전환하는 식이 있었는데 제거하면서 stream도 없앤다는걸 깜박했었네요 😢 감사합니다 ㅎㅎ